### PR TITLE
improve sprite init from score

### PR DESCRIPTION
**Pull request by matthewdean**
Changes:
- load behaviors correctly from score
- load correct sprite init data for the given frame
- in the frame loop, advance the frame regardless of whether it has been skipped (please double-check this is desired)
- support `go(frame_label)` instead of just `go(frame_num)`
- switch from setting sprite `cast_lib` and `cast_member` properties to calling `sprite_set_prop`

### CODE REVIEW

**Comment by matthewdean** [comment]
Now also deserializing film loops, which use the same memory format as score chunks.

In the cast member specific data section there are many fields which are present for film loops but which appear to be only relevant to other cast member types such as movies. I may see about generalizing the parsing code so it works for all.

File: vm-rust/src/player/handlers/datum_handlers/script.rs [hunk 1]
```diff
@@ -39,6 +39,7 @@ impl ScriptDatumHandlers {
 
   pub fn create_script_instance(script_ref: &CastMemberRef) -> (ScriptInstanceRef, DatumRef) {
     reserve_player_mut(|player| {
+      // TODO we are allocating two script instance ids here, should only be one
```
Comment (igorlira) [inline vm-rust/src/player/handlers/datum_handlers/script.rs:42]: Are you referring to `get_free_script_instance_id`? It doesn't allocate an instance id, only returns the next free id.

**Review by igorlira — APPROVED** [review]
Great stuff, thanks for sending this!

File: vm-rust/src/player/handlers/datum_handlers/script.rs (hunk 1 above)
Comment (matthewdean) [reply vm-rust/src/player/handlers/datum_handlers/script.rs:42]: Yes - to be more precise we're calling `get_free_script_instance_id` twice: once here and then inside `alloc_script_instance`. I assume we only want to call it once per script instance creation.

Comment (matthewdean) [reply vm-rust/src/player/handlers/datum_handlers/script.rs:42]: Thinking I should I change `alloc_script_instance` to use the `id` field of ScriptInstance or explicitly accept an id argument.

**Comment by igorlira** [comment]
I had to merge this manually due to conflicts so I'm closing the PR, but I've checked that all of its commits have been merged into main.

**Pull request closed without merging**